### ISSUE TITLE
Knip says class members are unused but they are used

### ISSUE THREAD

**Issue by Zamiell** [webpro/knip#339]
This is on knip version 2.40.1.

Steps to reproduce:

```
git clone git@github.com:IsaacScript/isaacscript.git
cd isaacscript
git checkout aa5b9e68ca3ec147d0b87ec83da8bd8a536dcf61
yarn
npx knip
```

The final lines of the output are:

```
Unused exported class members (6)
$schema                              Config  packages/isaacscript-cli/src/classes/Config.ts
customTargetModDirectoryName         Config  packages/isaacscript-cli/src/classes/Config.ts
enableIsaacScriptWatcherAutoRestart  Config  packages/isaacscript-cli/src/classes/Config.ts
isaacScriptCommonDev                 Config  packages/isaacscript-cli/src/classes/Config.ts
modsDirectory                        Config  packages/isaacscript-cli/src/classes/Config.ts
saveSlot                             Config  packages/isaacscript-cli/src/classes/Config.ts
```

I think this is a bug, because these fields do appear to be being used. For example, let's consider "customTargetModDirectoryName". If we search for that string in VSCode, we see that it is being used in the `getModTargetDirectoryName` function in the `utils.ts` file. And if we look where that function is being used, we can very quickly work our way back to the "main.ts" file, which is explicitly listed in the "entry" field in the "knip.json" file.

Furthermore, there are other bugs contained in this output that do not have to do with class members, do you want me to open separate issues for those?

**Comment by webpro** [comment]
Thanks for creating the reproduction, very clear. I looked into this today, but unfortunately it seems not straightforward to fix.

When running with `knip -W packages/isaacscript-cli --production --strict` the class members are not reported as unused, so the functionality itself works fine. Yet with multiple workspaces somehow the moment it tries to find references to the class members, not all files are resolved cq part of the program (so it doesn't find references). Will need to dig deeper.

And yes, please file one issue per bug. This way, we can discuss and keep track of them separately. Some take more time than others. I can close them separately when finished.

**Comment by webpro** [comment]
The issue was a duplicate package name, which Knip will now throw for.

**Comment by webpro** [comment]
:rocket: _This issue has been resolved in v3.1.0. See [Release 3.1.0](https://github.com/webpro/knip/releases/tag/3.1.0) for release notes._
